### PR TITLE
feat: `deno2node --init`

### LINKED ISSUE #20: Be a `tsc`.
I would love `deno2node` to behave exactly like a `tsc` that is Deno-aware. I see this as the main advantage over `dnt`—instead of having some magic black box that somehow tries to do everything, it would be awesome to do just one thing, and do it really well: compiling Deno code for Node.

That way, we could use our existing Node workflows that are based around `tsc`, and use `deno2node` as a drop-in replacement once we move to Deno. As a result, all existing infra and Node tooling will keep on working. I cannot imagine how `dnt` is going to solve https://github.com/denoland/dnt/issues/60 in the foreseeable future, but `deno2node` already integrates seamlessly with Node workflows.

At the same time, it enables us to write Deno code most of the time. We can use all their tooling, hosting, docs, infra, and all other advantages. On the Node side, in contrast to `dnt`, we still remain in control of our own configs and setup. In a way, `deno2node` brings the best of both worlds, because we can target both platforms natively, and use the familiar tooling of both ecosystems.

Some points that would be amazing to see are the following.
- `deno2node --init` creates a clean project from scratch which targets both Deno and Node.
- `deno2node` runs without a file path to `tsconfig.json`, and it behaves like `tsc` would do without config file.
- `deno2node` supports those compiler options that make sense from https://www.typescriptlang.org/docs/handbook/compiler-options.html#compiler-options

### PR DESCRIPTION
**Pull request by wojpawlik**
Fixes #20, but feel free to open follow-up pull requests.

### CODE REVIEW

File: src/init.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+}
+
+export async function initializeProject() {
+  const tsconfigUrl = new URL("../tsconfig.json", import.meta.url);
+  await Promise.all([
+    createShimFile(),
+    createPackageJson(),
+    download(tsconfigUrl, "tsconfig.json"),
+    Deno.writeTextFile(".gitignore", gitignore),
+  ]);
```
Comment (KnorpelSenf) [inline src/init.ts:45]: Disk IO is not exactly faster when done in parallel. (Not saying performance things would be measurable here.) Performing stuff sequentially leads to better errors.
```suggestion
  await createShimFile()
  await createPackageJson()
  await download(tsconfigUrl, "tsconfig.json")
  await Deno.writeTextFile(".gitignore", gitignore)
```
Note that it's actually quite common for CLIs to use the sync versions.

**Comment by wojpawlik** [comment]
On 2nd thought, template repo would probably be easier to maintain

**Comment by KnorpelSenf** [comment]
> On 2nd thought, template repo would probably be easier to maintain

Maybe. Either way, good tools focus on being _good for the user_, not by being _good for the developers_ by needing as little maintenance as possible. Having to clone a template repo (open browser, navigate to GitHub, find template, create repo, copy URL, clone repo) is much worse than doing `deno2node --init`.

If you are afraid that maintenance becomes a burden, it helps to have a more popular tool, as this increases the chances of other people helping out. In turn, I can offer to take the responsibility for the init files. If someone discovers a problem with the files and submits a fix, I can do the review and hit merge, so that you have as little work as possible :)

File: src/init.ts (hunk 1 above)
Comment (KnorpelSenf) [reply src/init.ts:45]: I think I did not consider that you're doing quite a lot of network IO when I wrote this.

**Comment by wojpawlik** [comment]
That's not it. Imperative code with hardcoded strings is inherently more difficult to understand, modify, and extend, than https://github.com/fromdeno/template.

A CLI that can download and unpack arbitrary tarball could quickly bootstrap a project, perhaps npm has one already?

**Comment by KnorpelSenf** [comment]
I see no contradiction. Why not provide that functionality behind `deno2node --init`?

**Comment by wojpawlik** [comment]
No point if standalone CLI can do it

**Comment by wojpawlik** [comment]
Also, `deno2node` is tricky to extend because it runs on both runtimes

**Comment by wojpawlik** [comment]
And running on Deno is the only means of testing the library

**Comment by KnorpelSenf** [comment]
Do I understand you correctly that easy project initialisation is out of scope for this tool?

**Comment by wojpawlik** [comment]
Yes, although if a separate library exists for it, `deno2node --init` could import it dynamically

**Review by KnorpelSenf — APPROVED** [review]
Not tested yet, but LGTM. I'll try it out in the coming days, too, but if it works for you, I'll consider it good enough.

**Pull request merged**